source/software/matlab_parallel_computing.rst

```diff
@@ -129,43 +105,15 @@ You can start an interactive job using the ``parpool`` function:
 ::
 
     >> c = parcluster;
-    >> p = parpool(72); % requesting 72 cores
```

Comment:
That is somewhat more complicated as it is something that Matlab determines on itself. Normally you would think that it just request 1 node and up to 72 cores with parpool(<=72) , but I have seen other scenarios where there are multiple nodes even when there are less than 72 cores. I suspect it just does a more general request, and receives whatever is available.

Reply:
But 'whatever is available' is determined by the srun/salloc command that is used, no? Assume for the sake of the argument that a user wants to use a parpool of 2. Here are two srun commands that could be used (executed on the sandbox):

$ srun --nodes=2   -A lpt2_sysadmin --pty bash -l
srun: job 60004811 queued and waiting for resources
srun: job 60004811 has been allocated resources
[...]
SLURM_NNODES: 2
SLURM_NODELIST: e25c03n[1-2]
SLURM_JOB_CPUS_PER_NODE: 1(x2)

versus

$ srun --nodes=1 --ntasks=2   -A lpt2_sysadmin --pty bash -l
[]
SLURM_NNODES: 1
SLURM_NODELIST: e25c03n1
SLURM_JOB_CPUS_PER_NODE: 2
Date: Tue Apr 30 08:30:43 CEST 2024
Walltime: 00-01:00:00

In both cases Matlab has two cpus at its disposal (I would think), but I would assume that having two cores on one node is better than having one core on each of two nodes. 

In a way my question is more general than parallel matlab, and this indeed is not the place to educate the users on that. But perhaps it may be useful to have a small comment that the matlab parpool uses the resources available, and that these available resources are determined by the options of srun/salloc? At least if that is what happens :-).

Reply:
It can be both scenarios. I am pretty sure that this is caused by a not so good implementation by Mathworks, and I am not so keen on complicating this matter for our users by trying to explain this with what happens in the background. I completely agree that this is messy, but I think it is better that we take this up with Mathworks and let them revise our parallel server scripts (which are partially custom-made for us by them).